pkg/matcher/cel.go

```diff
@@ -162,3 +185,135 @@ func (t celPac) CompileOptions() []cel.EnvOption {
 				cel.UnaryBinding(t.pathChanged))),
 	}
 }
+
+// walkExprForLabelReferences recursively walks a CEL expression AST to find
+// references to event_type or label-related fields. Returns true if the expression
+// references "event_type" variable or ".labels" field access.
+//
+// CEL AST Node Types:
+//   - IdentExpr: Variable references (e.g., "event_type", "body")
+//   - SelectExpr: Field access with dot notation (e.g., body.labels, body.pull_request.labels)
+//   - CallExpr: Function calls and operators (e.g., size(), exists(), _[_] for bracket notation)
+//   - ListExpr: List literals (e.g., [1, 2, 3])
+//   - StructExpr: Map/struct literals (e.g., {"key": "value"})
+//   - ComprehensionExpr: List operations (e.g., list.exists(x, condition), list.filter(...))
+func walkExprForLabelReferences(expr *exprpb.Expr) bool {
+	if expr == nil {
+		return false
+	}
+
+	switch e := expr.GetExprKind().(type) {
+	case *exprpb.Expr_ConstExpr:
+		// ConstExpr: Literal constant values (strings, numbers, bools, null, bytes)
+		// Constants cannot contain label references.
+		return false
+
+	case *exprpb.Expr_IdentExpr:
+		// IdentExpr: A simple variable/identifier reference
+		// Examples: "event_type", "body", "event", "target_branch"
+		// We check if the user is referencing "event_type" to handle label events explicitly.
+		if e.IdentExpr.GetName() == "event_type" {
+			return true
+		}
+
+	case *exprpb.Expr_SelectExpr:
+		// SelectExpr: Field access using dot notation (the "." operator)
+		// Examples:
+		//   - body.labels           -> SelectExpr{Operand: body, Field: "labels"}
+		//   - body.pull_request     -> SelectExpr{Operand: body, Field: "pull_request"}
+		//   - x.name                -> SelectExpr{Operand: x, Field: "name"}
+		// We check if the field being accessed is "labels".
+		if e.SelectExpr.GetField() == "labels" {
```

Comment:
so what about this then, so gitlab gets supported
```suggestion
 	field := e.SelectExpr.GetField()
	if field == "labels" || field == "pull_request_labels" {
```

Reply:
Added `pull_request_labels` support in 7ad611cfdb46b35d8d6a12cc1e1bbc622295797a